flash/text/generation/model.py

```diff
@@ -0,0 +1,98 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import warnings
+from typing import Optional, Union, Callable, Mapping, Type, Sequence, List, Dict, Any
+
+from pytorch_lightning import Callback
+from torchmetrics import Metric
+import torch
+from flash.core.utilities.imports import _TEXT_AVAILABLE
+
+from flash.core.model import Task
+from flash.text.ort_callback import ORTCallback
+
+if _TEXT_AVAILABLE:
+    from transformers import GPT2Model, PreTrainedTokenizerBase
+    from transformers.models.gpt2.modeling_gpt2 import GPT2DoubleHeadsModelOutput
+
+else:
+    GPT2Model, PreTrainedTokenizerBase = None, None
+
+class TextGeneration(Task):
+    """
+
+    """
+    required_extras: str = "text"
+
+    def __init__(
+            self,
+            backbone: str = "gpt2",
+            loss_fn: Optional[Union[Callable, Mapping, Sequence]] = None,
+            optimizer: Type[torch.optim.Optimizer] = torch.optim.Adam,
+            metrics: Union[Metric, Callable, Mapping, Sequence, None] = None,
+            learning_rate: float = 5e-5,
+            val_target_max_length: Optional[int] = None,
+            num_beams: Optional[int] = None,
+            enable_ort: bool = False,
+    ):
+        os.environ["TOKENIZERS_PARALLELISM"] = "TRUE"
+        # disable HF thousand warnings
+        warnings.simplefilter("ignore")
+        # set os environ variable for multiprocesses
+        os.environ["PYTHONWARNINGS"] = "ignore"
+        super().__init__(loss_fn=loss_fn, optimizer=optimizer, metrics=metrics, learning_rate=learning_rate)
+        self.model = GPT2Model.from_pretrained(backbone)
```

Comment:
Could we instead use ``AutoModelWithLMHead`` here rather than limiting to GPT2? Might need to add support for padding text for other models to work but at least we can have the option like here: https://huggingface.co/transformers/usage.html#text-generation